```python
python ./tools/codestyle/clang-tidy.py -p=build -j=10 -clang-tidy-binary=clang-tidy -extra-arg=-Wno-unknown-warning-option -extra-arg=-Wno-pessimizing-move -extra-arg=-Wno-braced-scalar-init -extra-arg=-Wno-deprecated-copy -extra-arg=-Wno-dangling-gsl -extra-arg=-Wno-final-dtor-non-final-class -extra-arg=-Wno-implicit-int-float-conversion -extra-arg=-Wno-inconsistent-missing-override -extra-arg=-Wno-infinite-recursion -extra-arg=-Wno-mismatched-tags  -extra-arg=-Wno-self-assign -extra-arg=-Wno-sign-compare -extra-arg=-Wno-sometimes-uninitialized -extra-arg=-Wno-tautological-overlap-compare -extra-arg=-Wno-unused-const-variable -extra-arg=-Wno-unused-lambda-capture -extra-arg=-Wno-unused-private-field -extra-arg=-Wno-unused-value -extra-arg=-Wno-unused-variable  -extra-arg=-Wno-overloaded-virtual  -extra-arg=-Wno-defaulted-function-deleted  -extra-arg=-Wno-delete-non-abstract-non-virtual-dtor  -extra-arg=-Wno-return-type-c-linkage  > clang-tidy_v2.log  2>&1 
```

运行完还会有相关报错

```
Resource filename: /usr/local/lib/python3.11/dist-packages/clang_tidy/data/bin/clang-tidy 
clang-tidy -extra-arg=-Wno-unknown-warning-option -extra-arg=-Wno-pessimizing-move -extra-arg=-Wno-braced-scalar-init -extra-arg=-Wno-deprecated-copy -extra-arg=-Wno-dangling-gsl -extra-arg=-Wno-final-dtor-non-final-class -extra-arg=-Wno-implicit-int-float-conversion -extra-arg=-Wno-inconsistent-missing-override -extra-arg=-Wno-infinite-recursion -extra-arg=-Wno-mismatched-tags -extra-arg=-Wno-self-assign -extra-arg=-Wno-sign-compare -extra-arg=-Wno-sometimes-uninitialized -extra-arg=-Wno-tautological-overlap-compare -extra-arg=-Wno-unused-const-variable -extra-arg=-Wno-unused-lambda-capture -extra-arg=-Wno-unused-private-field -extra-arg=-Wno-unused-value -extra-arg=-Wno-unused-variable -extra-arg=-Wno-overloaded-virtual -extra-arg=-Wno-defaulted-function-deleted -extra-arg=-Wno-delete-non-abstract-non-virtual-dtor -extra-arg=-Wno-return-type-c-linkage -p=build /paddle/paddle/testing/paddle_gtest_main.cc
/paddle/third_party/eigen3/unsupported/Eigen/CXX11/../../../Eigen/Core:70:10: error: 'omp.h' file not found [clang-diagnostic-error]
#include <omp.h>
         ^~~~~~~
Resource filename: /usr/local/lib/python3.11/dist-packages/clang_tidy/data/bin/clang-tidy 
clang-tidy -extra-arg=-Wno-unknown-warning-option -extra-arg=-Wno-pessimizing-move -extra-arg=-Wno-braced-scalar-init -extra-arg=-Wno-deprecated-copy -extra-arg=-Wno-dangling-gsl -extra-arg=-Wno-final-dtor-non-final-class -extra-arg=-Wno-implicit-int-float-conversion -extra-arg=-Wno-inconsistent-missing-override -extra-arg=-Wno-infinite-recursion -extra-arg=-Wno-mismatched-tags -extra-arg=-Wno-self-assign -extra-arg=-Wno-sign-compare -extra-arg=-Wno-sometimes-uninitialized -extra-arg=-Wno-tautological-overlap-compare -extra-arg=-Wno-unused-const-variable -extra-arg=-Wno-unused-lambda-capture -extra-arg=-Wno-unused-private-field -extra-arg=-Wno-unused-value -extra-arg=-Wno-unused-variable -extra-arg=-Wno-overloaded-virtual -extra-arg=-Wno-defaulted-function-deleted -extra-arg=-Wno-delete-non-abstract-non-virtual-dtor -extra-arg=-Wno-return-type-c-linkage -p=build /paddle/paddle/testing/paddle_gtest_main.cc
/paddle/third_party/eigen3/unsupported/Eigen/CXX11/../../../Eigen/Core:70:10: error: 'omp.h' file not found [clang-diagnostic-error]
#include <omp.h>
```

```
Resource filename: /usr/local/lib/python3.11/dist-packages/clang_tidy/data/bin/clang-tidy 
clang-tidy -extra-arg=-Wno-unknown-warning-option -extra-arg=-Wno-pessimizing-move -extra-arg=-Wno-braced-scalar-init -extra-arg=-Wno-deprecated-copy -extra-arg=-Wno-dangling-gsl -extra-arg=-Wno-final-dtor-non-final-class -extra-arg=-Wno-implicit-int-float-conversion -extra-arg=-Wno-inconsistent-missing-override -extra-arg=-Wno-infinite-recursion -extra-arg=-Wno-mismatched-tags -extra-arg=-Wno-self-assign -extra-arg=-Wno-sign-compare -extra-arg=-Wno-sometimes-uninitialized -extra-arg=-Wno-tautological-overlap-compare -extra-arg=-Wno-unused-const-variable -extra-arg=-Wno-unused-lambda-capture -extra-arg=-Wno-unused-private-field -extra-arg=-Wno-unused-value -extra-arg=-Wno-unused-variable -extra-arg=-Wno-overloaded-virtual -extra-arg=-Wno-defaulted-function-deleted -extra-arg=-Wno-delete-non-abstract-non-virtual-dtor -extra-arg=-Wno-return-type-c-linkage -p=build /paddle/paddle/phi/kernels/funcs/segment_pooling.cu
error: no such file or directory: 'arch=compute_89,code=sm_89' [clang-diagnostic-error]
error: too many errors emitted, stopping now [clang-diagnostic-error]
error: unknown argument: '-forward-unknown-to-host-compiler' [clang-diagnostic-error]
error: unknown argument: '-gencode' [clang-diagnostic-error]
error: unsupported option '--expt-extended-lambda' [clang-diagnostic-error]
error: unsupported option '--expt-relaxed-constexpr' [clang-diagnostic-error]
/paddle/paddle/phi/kernels/funcs/gather.cu.h:60:7: error: cannot use 'throw' in __global__ function [clang-diagnostic-error]
```

--- a/tools/codestyle/clang-tidy.py
+++ b/tools/codestyle/clang-tidy.py
@@ -127,14 +127,17 @@ def skip_check_file(database, build_path):
     skip_file_list.append(".cu")
     skip_file_list.append(os.path.join(os.getcwd(), build_path))
     skip_file_list += analysis_gitignore(os.getcwd())
+    res_list = []
     for entry in database:
+        write_in = True
         for ignore_file in skip_file_list:
             if ignore_file in entry["file"]:
-                try:
-                    database.remove(entry)
-                except ValueError:
-                    pass
-    return database
+                write_in = False
+                break
+        if write_in:
+            res_list.append(entry)
+
+    return res_list
 
 
 def get_tidy_invocation(

--- a/tools/codestyle/clang-tidy.py
+++ b/tools/codestyle/clang-tidy.py
@@ -95,30 +95,29 @@ def make_absolute(f, directory):
 
 def analysis_gitignore(path, filename=".gitignore"):
     """Analysis gitignore file and return ignore file list"""
-    f = open(path + "/" + filename, "r")
-    lines = f.readlines()
-    ignore_file_list = []
-    for line in lines:
-        # Blank row
-        if line == "\n":
-            continue
+    with open(path + "/" + filename, "r") as f:
+        lines = f.readlines()
+        ignore_file_list = []
+        for line in lines:
+            # Blank row
+            if line == "\n" or line == "\r\n":
+                continue
 
-        # explanatory note
-        line = line.replace("\n", "")
-        if "#" in line:
-            if line[0] != "#":
-                ignore_file_list.append(
-                    line[: line.index("#")].replace(" ", "")
-                )
-            continue
+            # explanatory note
+            line = line.replace("\n", "").strip()
+            if "#" in line:
+                if not line.startswith("#"):
+                    ignore_file_list.append(
+                        line[: line.index("#")].replace(" ", "")
+                    )
+                continue
 
-        # TODO(gouzil): support more gitignore rules
-        if "*" in line:
-            continue
+            # TODO(gouzil): support more gitignore rules
+            if "*" in line:
+                continue
 
-        ignore_file_list.append(line.replace(" ", ""))
+            ignore_file_list.append(line.replace(" ", ""))
 
-    f.close()
     return ignore_file_list
 
 

